File: library-extensions/src/main/java/com/mikepenz/fastadapter_extensions/adapters/WrapAdapter.java
package com.mikepenz.fastadapter_extensions.adapters;

import com.mikepenz.fastadapter.IItem;
import com.mikepenz.fastadapter.commons.adapters.AbstractWrapAdapter;

import java.util.List;

/**
 * Created by mikepenz on 03.03.16.
 */
public class WrapAdapter<Item extends IItem> extends AbstractWrapAdapter<Item> {
    public WrapAdapter(List<Item> items) {
        super(items);
    }

    @Override
    public void setItems(List<Item> items) {
        super.setItems(items);
        notifyDataSetChanged();
    }

    @Override
    public boolean shouldInsertItemAtPosition(int position) {
        if (getAdapter().getItemCount() > 0 && getItems().size() > 0) {
            int itemsInBetween = (getAdapter().getItemCount() + getItems().size()) / (getItems().size() + 1) + 1;
            return (position + 1) % itemsInBetween == 0;
        }
        return false;
    }

    @Override
    public int itemInsertedBeforeCount(int position) {
        if (getAdapter().getItemCount() > 0 && getItems().size() > 0) {
            int itemsInBetween = (getAdapter().getItemCount() + getItems().size()) / (getItems().size() + 1) + 1;
            return position / itemsInBetween;
        }
        return 0;
    }
}

File: fastadapter-extensions-utils/src/main/java/com/mikepenz/fastadapter/adapters/WrapAdapter.kt
package com.mikepenz.fastadapter.adapters

import androidx.recyclerview.widget.RecyclerView
import com.mikepenz.fastadapter.IItem

/**
 * Created by mikepenz on 03.03.16.
 */
class WrapAdapter<Item : IItem<VH>, VH : RecyclerView.ViewHolder>(items: List<Item>) : AbstractWrapAdapter<Item, VH>(items) {

    override var items: List<Item> = emptyList()
        set(items) {
            field = items
            notifyDataSetChanged()
        }

    override fun shouldInsertItemAtPosition(position: Int): Boolean {
        if (adapter?.itemCount ?: 0 > 0 && items.isNotEmpty()) {
            val itemsInBetween = (adapter?.itemCount ?: 0 + items.size) / (items.size + 1) + 1
            return (position + 1) % itemsInBetween == 0
        }
        return false
    }

    override fun itemInsertedBeforeCount(position: Int): Int {
        if (adapter?.itemCount ?: 0 > 0 && items.isNotEmpty()) {
            val itemsInBetween = (adapter?.itemCount ?: 0 + items.size) / (items.size + 1) + 1
            return position / itemsInBetween
        }
        return 0
    }
}
